cmd/tt/bucket.go

```diff
@@ -256,110 +306,131 @@ var bucketPathLinksCmd = &cobra.Command{
-	if len(rep.Item.Items) > 0 {
-		items = rep.Item.Items
-	} else if !rep.Item.IsDir {
-		items = append(items, rep.Item)
-	}
+		ctx, cancel := threadCtx(cmdTimeout)
+		defer cancel()
+		key := configViper.GetString("key")
+		rep, err := buckets.ListPath(ctx, key, pth)
+		if err != nil {
+			cmd.Fatal(err)
+		}
+		var items []*pb.ListPathReply_Item
+		if len(rep.Item.Items) > 0 {
+			items = rep.Item.Items
+		} else if !rep.Item.IsDir {
+			items = append(items, rep.Item)
+		}
 
-	var data [][]string
-	if len(items) > 0 && !strings.HasPrefix(pth, ".textile") {
-		for _, item := range items {
-			if item.Name == ".textile" {
-				continue
-			}
-			var links string
-			if item.IsDir {
-				links = strconv.Itoa(len(item.Items))
-			} else {
-				links = "n/a"
+		var data [][]string
+		if len(items) > 0 && !strings.HasPrefix(pth, configDir) {
+			for _, item := range items {
+				if item.Name == configDir {
+					continue
+				}
+				var links string
+				if item.IsDir {
+					links = strconv.Itoa(len(item.Items))
+				} else {
+					links = "n/a"
+				}
+				data = append(data, []string{
+					item.Name,
+					strconv.Itoa(int(item.Size)),
+					strconv.FormatBool(item.IsDir),
+					links,
+					item.Path,
+				})
 			}
-			data = append(data, []string{
-				item.Name,
-				strconv.Itoa(int(item.Size)),
-				strconv.FormatBool(item.IsDir),
-				links,
-				item.Path,
-			})
 		}
-	}
 
-	if len(data) > 0 {
-		cmd.RenderTable([]string{"name", "size", "dir", "items", "path"}, data)
-	}
-	cmd.Message("Found %d items", aurora.White(len(data)).Bold())
+		if len(data) > 0 {
+			cmd.RenderTable([]string{"name", "size", "dir", "objects", "path"}, data)
+		}
+		cmd.Message("Found %d objects", aurora.White(len(data)).Bold())
+	},
 }
 
-var pushBucketPathCmd = &cobra.Command{
-	Use:   "push [target] [path]",
-	Short: "Push to a bucket path (interactive)",
-	Long: `Push files and directories to a bucket path. Existing paths will be overwritten. Non-existing paths will be created.
-
-Using the '--org' flag will create a new bucket under the organization's account.
-
-File structure is mirrored in the bucket. For example, given the directory:
-    foo/one.txt
-    foo/bar/two.txt
-    foo/bar/baz/three.txt
-
-These 'push' commands result in the following bucket structures.
-
-'tt bucket push foo/ .':
-    one.txt
-    bar/two.txt
-    bar/baz/three.txt
-		
-'tt bucket push foo mybuck':
-    mybuck/foo/one.txt
-    mybuck/foo/bar/two.txt
-    mybuck/foo/bar/baz/three.txt
-
-'tt bucket push foo/bar mybuck':
-    mybuck/bar/two.txt
-    mybuck/bar/baz/three.txt
-
-'tt bucket push foo/bar/baz mybuck':
-    mybuck/baz/three.txt
-
-'tt bucket push foo/bar/baz/three.txt mybuck':
-    mybuck/three.txt
-`,
-	Args: cobra.MinimumNArgs(2),
+var bucketPushCmd = &cobra.Command{
+	Use:   "push",
+	Short: "Push bucket object changes",
+	Long:  `Pushes paths that have been added to and paths that have been removed or differ from the local bucket root.`,
```

Comment:
^ what's happening. There's a lot to explain.. not gonna try to do it here. Hopefully the code is somewhat readable.